analyses/independent-samples-release/util/independent-rna-samples.R

```diff
@@ -0,0 +1,118 @@
+#' "primary_plus" if you would like to select other non-initial tumor RNA-Seq sample if no 
+#' initial tumor RNA-Seq sample exists
+#' or "Diagnosis" in the `tumor_descriptor` field.
+#' @param seed An optional random number seed. 
+#' 
+#' @return a data frame of Participant and Specimen IDs, each present only once.
+independent_rna_samples <- function(independent_dna_sample_df, 
+                                histology_df,
+                                match_type = c("independent_dna", "independent_dna_plus_only_rna"),
+                                tumor_description_rna_only = c("primary","primary_plus"),
+                                seed){
+  match_type <- match.arg(match_type)
+  tumor_description_rna_only <- match.arg(tumor_description_rna_only)
+  if(!missing(seed)){set.seed(seed)}
+  primary_descs <- c("Initial CNS Tumor", "Diagnosis", "Primary Tumor")
+  
+  
+  
+  # Find sample set for the dna independent samples 
+  # This will always be the included since in both the following
+  # conditions "independent_dna" "independent_dna_plus_only_rna"
+  # 
+  independent_dna <- histology_df %>%
+    dplyr::mutate(match_id = paste(Kids_First_Participant_ID, sample_id, sep = "_")) %>%
+    # include matched independent_dna samples
+    dplyr::filter(Kids_First_Biospecimen_ID %in%
+                    independent_dna_sample_df$Kids_First_Biospecimen_ID)
+  matched_rna <- histology_df %>%
+    dplyr::mutate(match_id = paste(Kids_First_Participant_ID, sample_id, sep = "_"),
+                  age_at_diagnosis_days = as.numeric(age_at_diagnosis_days)) %>%
+    # keep rna from histology_df
+    dplyr::filter(experimental_strategy == "RNA-Seq",
+                  # find participants which have matching dna samples in independent_wgswxs
+                  Kids_First_Participant_ID %in% independent_dna$Kids_First_Participant_ID,
+                  # keep specific sample_ids since some participants might have multiple sample_ids
+                  match_id %in% independent_dna$match_id)
+  
+  # has rna samples which match the independent samples provided
+  sample_df <- matched_rna
+
+  
+  # Here we are adding only initial only-RNA-Seq samples
+  # since this will always to part of independent_dna_plus_only_rna
+  # regardless tumor_description_rna_only is "primary" OR "primary_plus"
+  #
+  if( match_type == "independent_dna_plus_only_rna" ) {
+    # find sample set where we initial only-RNA-Seq samples
+    only_rna_intial <- histology_df %>% 
+      # keep rna from histology_df
+      dplyr::filter(experimental_strategy == "RNA-Seq",
+                    tumor_descriptor %in% primary_descs,
+                    # find and remove participants which have 
+                    # matching dna samples in independent_wgswxs 
+                    !Kids_First_Participant_ID %in% independent_dna$Kids_First_Participant_ID)
```

Comment:
```suggestion
                    !Kids_First_Participant_ID %in% sample_df$Kids_First_Participant_ID)
```

I am not sure about this suggestion. I was considering whether it is possible to have `independent_dna$Kids_First_Participant_ID`s that have no matching sample.

If an `independent_dna$Kids_First_Participant_ID` has no matching RNA-seq sample, should all its primary RNA-seq samples still be kept? However, such RNA-seq samples will not be RNA-seq only sample, but rather they are RNA-seq samples that have no matched DNA sample.

Reply:
Agree with @logstar logic here. Should be checking against `sample_df`  which at this step has matched rna (both primary and non-primary)